bundles/org.openhab.binding.konnected/src/main/java/org/openhab/binding/konnected/internal/gson/KonnectedModuleGson.java

```diff
@@ -36,12 +41,50 @@
     private Integer pollInterval;
     private String addr;
 
-    public Integer getPin() {
-        return pin;
+    private Integer getPin() {
+
+        return Arrays.asList(PIN_TO_ZONE).indexOf(pin);
     }
 
-    public void setPin(Integer setPin) {
-        this.pin = setPin;
+    private void setPin(Integer setPin) {
```

Comment:
Primitive types should be used where possible.
```suggestion
    private void setPin(int setPin) {
```